.github/actions/restore_dbt_cache/action.yaml

```diff
@@ -0,0 +1,76 @@
+name: Restore dbt cache
+description: Attempts to restore dbt cache from S3 storage.
+inputs:
+  path:
+    description: The local path to restore the cache to in case of a hit.
+    required: true
+  key:
+    description: The cache key to query for.
+    required: true
+  restore-key:
+    description: An additional key to use as a fallback for the cache.
+    required: true
+  bucket:
+    description: The S3 bucket that should store the cache.
+    required: false
+    default: ccao-dbt-cache-us-east-1
+outputs:
+  cache-hit:
+    description: >-
+      Boolean indicating whether a match was found for the cache key.
+    value: ${{ steps.query-cache-keys.outputs.cache-hit }}
+  exact-match:
+    description: >-
+      Boolean indicating whether a cache hit was an exact match. Always
+      false if cache-hit is false.
+    value: ${{ steps.query-cache-keys.outputs.exact-match }}
+  cache-matched-key:
+    description: The cache key that matched, if any. Empty if cache-hit is false.
+    value: ${{ steps.query-cache-keys.outputs.cache-matched-key }}
+runs:
+  using: composite
+  steps:
+    # Check if there is an exact or fallback match for the cache key
+    - name: Check for a cache key match
+      id: query-cache-keys
+      run: |
+        if aws s3api head-object --bucket "$BUCKET" --key "$KEY/manifest.json"; then
```

Comment:
Note that this simple `if ... then` conditional will fail on _any_ nonzero status from `aws s3api head-object`, not just the expected 404. I think this is acceptable, since we want to continue without the cache in case of an unexpected error (e.g. a 403 forbidden due to bad credentials), and [testing confirms](https://github.com/ccao-data/data-architecture/actions/runs/5965925106/job/16184439121#step:4:79) that the `aws s3api head-object` call will print the exit status to the console in these cases so we should be able to debug them.

```
An error occurred (403) when calling the HeadObject operation: Forbidden
```

Reply:
**praise:** Nice, this is perfect. Super nice simplifications all around.